src/Starcounter.Database.Extensions/NestedTransactor.cs

```diff
@@ -114,47 +114,52 @@ public override async Task<T> TransactAsync<T>(Func<IDatabaseContext, T> functio
             }
         }
 
-        public override Task TransactAsync(Func<IDatabaseContext, Task> function, TransactOptions options = null)
+        public override async Task TransactAsync(Func<IDatabaseContext, Task> function, TransactOptions options = null)
         {
             IDatabaseContext nested = _current.Value;
 
             if (nested == null)
             {
-                return base.TransactAsync(function, options);
+                await base.TransactAsync(function, options);
             }
-
-            var context = new NestedTransactionContext(nested);
-
-            try
-            {
-                return function(context);
-            }
-            catch (Exception e)
+            else
             {
-                Rollback(context);
-                return Task.FromException(e);
+                var context = new NestedTransactionContext(nested);
+
+                try
+                {
+                    await function(context);
+                    return;
```

Comment:
not needed `return`

Reply:
Ha, that one I just dropped locally _before_ you mentioned it. :)